prometheus/promhttp/http.go

```diff
@@ -155,29 +145,50 @@ func HandlerFor(reg prometheus.Gatherer, opts HandlerOpts) http.Handler {
 				}
 			}
 		}
-		if closer, ok := writer.(io.Closer); ok {
-			closer.Close()
-		}
+
 		if lastErr != nil && buf.Len() == 0 {
 			http.Error(w, "No metrics encoded, last error:\n\n"+lastErr.Error(), http.StatusInternalServerError)
 			return
 		}
-		header := w.Header()
-		header.Set(contentTypeHeader, string(contentType))
-		header.Set(contentLengthHeader, fmt.Sprint(buf.Len()))
```

Comment:
This commit removes this header (which SHOULD be used as per https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.13 ). That's probably OK for gzip'd case because we avoid another buffering layer, but it could still be set in the cases where we don't compress. This is complicated here because of the double-wrapping, see comment below.